src/main/java/fruitshop/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
```

Comment:
The check for `newQuantity >= 0` is unnecessary for the `SupplyOperation`. Supply operations are expected to increase the quantity of fruit in the storage, so it is not possible for the new quantity to be negative. You can remove this check.